testing/mock/test_system.cpp

```diff
@@ -206,17 +207,18 @@ test_system *test_system::instance() {
   return test_system::instance_;
 }
 
-std::string test_system::prepare_syfs(const test_platform &platform) {
-  std::string tmpsysfs = "tmpsysfs-XXXXXX";
+void test_system::prepare_syfs(const test_platform &platform) {
+  char tmpsysfs[]{ "tmpsysfs-XXXXXX\0" };
```

Comment:
The explicit \0 is redundant.  C terminate all string constants with \0.

Reply:
I was just testing you...